fineract-core/src/main/java/org/apache/fineract/infrastructure/core/service/database/DatabaseSpecificSQLGenerator.java

```diff
@@ -189,4 +207,49 @@ public String castJson(String sql) {
             throw new IllegalStateException("Database type is not supported for casting to json " + databaseTypeResolver.databaseType());
         }
     }
+
+    public static String alias(@NotNull String field, String alias) {
```

Comment:
How are these methods below specific to the database type?

Reply:
Just looking at all these functions in general (not only Marta's): this is getting more and more complex... a fight we cannot win.... and in the end we have compatibility for 2 database systems. A lot of work (also future maintenance). Why not hit the breaks here and use e. g. JOOQ. Slowly, but steadily we are recreating it here with less features and less supported databases. For another day to decide, but I think it will be an important improvement.... and all this code can go away.